to-do-list-backend/src/api/express.ts

```diff
@@ -54,12 +54,36 @@ export function createServer(dependencies: Dependencies): Express {
         resp.send({});
     })
 
-    app.get('/api/tasks', (_, resp: Response) => {
+    app.get('/api/task', (_, resp: Response) => {
         const tasks = dependencies.storage.listTasks();
 
         resp.status(StatusCodes.OK);
         resp.send({ tasks: tasks })
     })
 
+    app.post('/api/task/mark-done', (req, res) => {
+        const { name } = req.body;
+
+        const task = tasks.find(task => task.name === name);
+        if (task) {
+            task.done = true;
```

Comment:
Is the task updated in storage? Try to call `/api/task/mark-done`, then `/api/tasks` – you will check if it works correctly.

You should probably create a new method `updateTask(task: Task)` in the `to-do-list-backend/src/storage/storage.ts` class and use it here. This method will iterate over `tasks` array with `map` method and update the task with equal name.

Like this:
```
updateTask(task: Task) {
    this.tasks.map((t: Task) => {
        if (t.name === task.name) {
            return task;
        }

        return t;
    })
}
```

Try this code out (I didn't check it so it may have some mistakes) – tweak it according to your requirements.

Reply:
i implemented the code in storage.ts, but how do i call it _/api/task/mark-done, then /api/tasks_?, with postman?

Reply:
Yes. Firstly, create a task calling POST /api/task, then call POST /api/task/mark-done, then GET /api/tasks via Postman